### PR TITLE
wifi: Fix stats crash deinit

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: drivers/wifi/nrf700x/src/net_if.c [hunk 1]
```diff
@@ -806,27 +837,38 @@ int nrf_wifi_if_set_config_zep(const struct device *dev,
...
 	}
 
-	/**
-	 * If vif_ctx_zep exists, rpu_ctx_zep and subsequent priv structures
-	 * should exist and cannot be NULL. Derive the pointers for further
-	 * structures
-	 */
+	ret = k_mutex_lock(&vif_ctx_zep->vif_lock, K_FOREVER);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to lock vif_lock", __func__);
```
Comment (jfischer-no) [inline drivers/wifi/nrf700x/src/net_if.c:848]: This is redundant and can be confusing if LOG_FUNC_NAME_PREFIX_ERR is used.

Comment (krish2718) [reply drivers/wifi/nrf700x/src/net_if.c:848]: Right, but this is prevalent across the driver, I have raised SHEL-2501Jira  to work on this outside this PR. This becomes tricky for OS agnostic code.

Comment (jfischer-no) [reply drivers/wifi/nrf700x/src/net_if.c:848]: > This becomes tricky for OS agnostic code.

This driver is for Zephyr RTOS in NCS, using Zephyr RTOS logging subsystem, or what was it implemented for?

Comment (krish2718) [reply drivers/wifi/nrf700x/src/net_if.c:848]: Yes, but this is only part of the driver, the other part is in `nrfxlib` which is OS agnostic. And this change need to be done for both, hence the Jira.

**Pull request merged**